File: src/main/java/mnm/mods/tabbychat/client/settings/GeneralServerSettings.java
package mnm.mods.tabbychat.client.settings;

import mnm.mods.tabbychat.util.ChannelPatterns;
import mnm.mods.tabbychat.util.MessagePatterns;
import mnm.mods.tabbychat.util.config.Value;
import mnm.mods.tabbychat.util.config.ValueList;
import mnm.mods.tabbychat.util.config.ValueObject;

public class GeneralServerSettings extends ValueObject {

    public Value<Boolean> channelsEnabled = value(true);
    public Value<Boolean> pmEnabled = value(true);
    public Value<ChannelPatterns> channelPattern = value(ChannelPatterns.BRACKETS);
    public Value<MessagePatterns> messegePattern = value(MessagePatterns.WHISPERS);
    public Value<Boolean> useDefaultTab = value(true);
    public ValueList<String> ignoredChannels = list();
    public Value<String> defaultChannel = value("");
    public Value<String> channelCommand = value("");
    public Value<String> messageCommand = value("");
}

File: src/main/kotlin/mnm/mods/tabbychat/client/settings/GeneralServerSettings.kt
package mnm.mods.tabbychat.client.settings

import mnm.mods.tabbychat.util.*

class GeneralServerSettings(config: AbstractConfigView, path: List<String>) : ConfigView(config, path) {

    @Comment("Enables channel detection on this server")
    var channelsEnabled by defining(true)
    @Comment("Enables direct message detection on this server")
    var pmEnabled by defining(true)
    @Comment("Specifies the hard-coded pattern to support channel detection")
    var channelPattern by definingEnum(ChannelPatterns.BRACKETS, ChannelPatterns::class.java)
    @Comment("Specifies the hard-coded pattern to support direct message detection")
    var messegePattern by definingEnum(MessagePatterns.WHISPERS, MessagePatterns::class.java)
    @Comment("Sends messages which detected as a channel to the default channel (*)")
    var useDefaultTab by defining(true)
    @Comment("The list of ignored channels")
    var ignoredChannels by definingList<String>()
    @Comment("The default channel name (not implemented)")
    var defaultChannel by defining("*")
//    var channelCommand by defining("")
//    var messageCommand by defining("")
}
